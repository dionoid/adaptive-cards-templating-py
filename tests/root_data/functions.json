{
    "title": "Main Title",
    "message": "{\"type\":\"Deployment\",\"buildId\":\"9542982\",\"releaseId\":\"129\",\"buildNumber\":\"20180504.3\",\"release\":{\"name\":\"Release-104\"},\"repoProvider\":\"GitHub\"}",
    "items": [
        {"title": "Item 1", "priceChange": 3},
        {"title": "Item 2", "priceChange": -4},
        {"title": "Item 3", "priceChange": 0}
    ],
    "strings": [
        "one",
        "two",
        "three"
    ]
}