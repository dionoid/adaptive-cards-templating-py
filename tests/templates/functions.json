{
    "type": "AdaptiveCard",
    "body": [
        {
            "type": "TextBlock",
            "text": "${json(message).release.name}"
        },
        {
            "$data": "${items}",
            "$when": "${if(priceChange == 0, false, true)}",
            "type": "TextBlock",
            "text": "${title} (${$index + 1}: ${$root.title}, host: ${$host.env})",
            "color": "${if(priceChange > 0, 'good', 'attention')}"
        }
    ]
}