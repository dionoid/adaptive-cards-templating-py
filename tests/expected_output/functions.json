{
    "type": "AdaptiveCard",
    "body": [
      {
        "type": "TextBlock",
        "text": "Release-104"
      },
      {
        "type": "TextBlock",
        "text": "Item 1 (1: Main Title, host: prod)",
        "color": "good"
      },
      {
        "type": "TextBlock",
        "text": "Item 2 (2: Main Title, host: prod)",
        "color": "attention"
      }
    ]
  }